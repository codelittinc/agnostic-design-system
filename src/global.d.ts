declare module '*.css';
declare module 'googlemaps';
declare module '*.svg' {
  const content: any;
  export default content;
}
