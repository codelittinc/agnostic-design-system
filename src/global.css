/* import font typeface */

@import "../node_modules/typeface-open-sans/index.css";

/* variables */

:root {
  --base-font-family: 'Open Sans', sans-serif;
  --base-font-size: 16px;
  --heading-1-font-size: 2.5rem;
  --heading-2-font-size: 2rem;
  --heading-3-font-size: 1.6rem;
  --heading-4-font-size: 1.5rem;
  --heading-5-font-size: 1.2rem;
  --heading-6-font-size: 1rem;
}

/* global styles */

* {
  box-sizing: border-box;
  -moz-box-sizing: border-box;
  -webkit-box-sizing: border-box;
}

body, input, textarea, button {
  font-family: var(--base-font-family);
}

p {
  font-size: var(--base-font-size);
}

h1 { font-size: var(--heading-1-font-size); }
h2 { font-size: var(--heading-2-font-size); }
h3 { font-size: var(--heading-3-font-size); }
h4 { font-size: var(--heading-4-font-size); }
h5 { font-size: var(--heading-5-font-size); }
h6 { font-size: var(--heading-6-font-size); }
