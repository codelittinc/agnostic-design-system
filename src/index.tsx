export { default as Button } from '@/components/Button';
export { default as Input } from '@/components/Input';
export { default as TextArea } from '@/components/TextArea';
export { default as SearchInput } from '@/components/SearchInput';
export { default as Dropdown } from '@/components/Dropdown';
export { default as List } from '@/components/List';
export { default as Toast } from '@/components/Toast';
export * from '@/components/Grid';
