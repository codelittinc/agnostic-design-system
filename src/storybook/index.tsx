export { default as Row } from './Row';
export { default as ColSection } from './ColSection';
