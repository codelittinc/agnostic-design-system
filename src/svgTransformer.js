const { process } = require('jest-svg-transformer');

module.exports = {
  process: (src, filename, config, options) => {
    const transformed = process(src, filename, config, options);
    return { code: transformed };
  }
};
