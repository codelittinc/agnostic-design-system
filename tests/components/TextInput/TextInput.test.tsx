import React from 'react'
import { cleanup } from '@testing-library/react'
import renderer from 'react-test-renderer'
import { TextInput } from '../../../src'

afterEach(cleanup)

describe('<TextInput />', () => {
  it('renders without crashing', () => {
    const tree = renderer.create(<TextInput name="batman_name" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('accepts a value', () => {
    const inputValue = 'Bruce Wayne'

    const tree = renderer
      .create(<TextInput name="batman_name" value={inputValue} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
