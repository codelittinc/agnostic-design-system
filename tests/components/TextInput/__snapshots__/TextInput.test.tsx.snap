// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> accepts a value 1`] = `
<input
  name="batman_name"
  type="text"
  value="Bruce Wayne"
/>
`;

exports[`<TextInput /> renders without crashing 1`] = `
<input
  name="batman_name"
  type="text"
/>
`;
